custom/aaa/views.py

```diff
@@ -46,18 +47,34 @@ def domain(self):
     def couch_user(self):
         return self.request.couch_user
 
+    @property
+    def user_ministry(self):
+        return self.couch_user.user_data.get('ministry')
+
+    def dispatch(self, *args, **kwargs):
+        if (not self.couch_user.is_web_user()
+                and (self.user_ministry is None or self.user_ministry == '')):
+            return no_permissions(self.request)
```

Comment:
This feels scary - I assume you're pretty sure this always exists for mobile users?  You might also want to add a `message` kwarg explaining what's going on.

Reply:
What about this feels scary? `user_data` should always be on the `couch_user` and I'm using `get` to ensure that there aren't any exceptions. They won't be on all mobile workers, only ones that have access to reporting dashboards. This in no way will affect any mobile app.

Dashboard users are setup as mobile workers. I'm not sure on the history of that with the CAS dashboard, but we are following the same format with AAA.

The default message for this is "Your user role allows you to access data based on your assigned location in the organization hierarchy. You do not currently have an assigned location, and will be unable to access CommCareHQ until that is corrected. Please contact your project administrator to be assigned to a location." because its a location_safe view. Its kind of a correct message in that the ministry needs to be set for the dashboard to know what location hierarchy to use for a mobile worker. I'm not going to change this for now since it would just add noise in transifex. If the ministry wants us to have a specific message in the future, they will provide one to be translated.

Reply:
> What about this feels scary?
Not the way you've written the code - the intended consequences.  Depends on how sure you are of the data in the system - if there are mobile workers currently using this feature who aren't configured properly, this could break it for them.  Not really sure what the current behavior is for such users anyways, though.

Reply:
there are no current users. since we are still pre-pilot and have not yet gone through QA